packages/gdpr-tool/src/main.js

```diff
@@ -0,0 +1,215 @@
+import { createClient } from '@commercetools/sdk-client'
+import { createRequestBuilder } from '@commercetools/api-request-builder'
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http'
+import {
+  createAuthMiddlewareForClientCredentialsFlow,
+  createAuthMiddlewareWithExistingToken,
+} from '@commercetools/sdk-middleware-auth'
+import { createUserAgentMiddleware } from '@commercetools/sdk-middleware-user-agent'
+import flatten from 'lodash.flatten'
+import silentLogger from './utils/silent-logger'
+import pkg from '../package.json'
+
+// todo add flow types
+export default class GDPRTool {
+  constructor(options) {
+    if (!options.apiConfig)
+      throw new Error('The constructor must be passed an `apiConfig` object')
+    this.apiConfig = options.apiConfig
+    this.client = createClient({
+      middlewares: [
+        createAuthMiddlewareWithExistingToken(
+          options.accessToken ? `Bearer ${options.accessToken}` : ''
+        ),
+        createAuthMiddlewareForClientCredentialsFlow(this.apiConfig),
+        createUserAgentMiddleware({
+          libraryName: pkg.name,
+          libraryVersion: pkg.version,
+        }),
+        createHttpMiddleware({ host: this.apiConfig.apiUrl }),
+      ],
+    })
+
+    this.logger = {
+      ...silentLogger,
+      ...options.logger,
+    }
+  }
+
+  getData(customerId) {
+    if (!customerId) throw Error('missing `customerId` argument')
+
+    this.logger.info('Starting fetch data')
+
+    const requestBuilder = createRequestBuilder({
+      projectKey: this.apiConfig.projectKey,
+    })
+    const customersUri = requestBuilder.customers
+      .where(`id = "${customerId}"`)
+      .perPage(500)
+      .build()
+    const ordersUri = requestBuilder.orders
+      .where(`customerId = "${customerId}"`)
+      .perPage(500)
+      .build()
+    const cartsUri = requestBuilder.carts
+      .where(`customerId = "${customerId}"`)
+      .perPage(500)
+      .build()
+    const paymentsUri = requestBuilder.payments
+      .where(`customer(id = "${customerId}")`)
+      .perPage(500)
+      .build()
+    const shoppingsListsUri = requestBuilder.shoppingLists
+      .where(`customer(id = "${customerId}")`)
+      .perPage(500)
+      .build()
+    const reviewsUri = requestBuilder.reviews
+      .where(`customer(id = "${customerId}")`)
+      .perPage(500)
+      .build()
+
+    const getUris = [
+      customersUri,
+      ordersUri,
+      cartsUri,
+      paymentsUri,
+      shoppingsListsUri,
+      reviewsUri,
+    ]
+
+    // todo change to process instead of execute
+    // in case of more than 500 results
```

Comment:
This quite important TODO. Will this be a part of this PR?

Reply:
Yup, added now! :-)